de.metas.business/src/main/java-legacy/org/compiere/process/MProductPriceCloningRequest.java

```diff
@@ -0,0 +1,78 @@
+package org.compiere.process;
+
+import static org.adempiere.model.InterfaceWrapperHelper.newInstance;
+
+import org.adempiere.ad.dao.IQueryBL;
+import org.adempiere.ad.trx.api.ITrx;
+import org.adempiere.mm.attributes.api.IAttributeDAO;
+import org.adempiere.model.InterfaceWrapperHelper;
+import org.adempiere.util.Services;
+import org.compiere.model.IQuery;
+import org.compiere.model.I_M_AttributeSetInstance;
+import org.compiere.model.I_M_ProductPrice;
+
+import lombok.Builder;
+
+/**
+ *
+ * @author metas-dev <dev@metasfresh.com>
+ *
+ */
+@Builder
+public class MProductPriceCloningRequest
+{
+	private final int source_PriceList_Version_ID;
+	private final int target_PriceList_Version_ID;
+
+	private final transient IAttributeDAO attributeDAO = Services.get(IAttributeDAO.class);
+
+	public void cloneProductPrice()
+	{
+		final IQueryBL queryBL = Services.get(IQueryBL.class);
+		final String trxName = ITrx.TRXNAME_None;
+
+		final IQuery<I_M_ProductPrice> existentProductPrices = queryBL.createQueryBuilder(I_M_ProductPrice.class, trxName)
+				.addEqualsFilter(I_M_ProductPrice.COLUMN_M_PriceList_Version_ID, target_PriceList_Version_ID)
+				.create();
+
+		queryBL.createQueryBuilder(I_M_ProductPrice.class, trxName)
+				.addEqualsFilter(I_M_ProductPrice.COLUMN_M_PriceList_Version_ID, source_PriceList_Version_ID)
+				.addNotInSubQueryFilter(I_M_ProductPrice.COLUMNNAME_M_Product_ID, I_M_ProductPrice.COLUMNNAME_M_Product_ID, existentProductPrices)
+				.create()
+				.iterateAndStream()
+				.forEach(this::createProductPrice);
+	}
+
+	private void createProductPrice(final I_M_ProductPrice productPrice)
+	{
+		final I_M_ProductPrice pp = newInstance(I_M_ProductPrice.class, productPrice);
```

Comment:
Please use org.adempiere.model.InterfaceWrapperHelper.copy() instead of manually copying it.
It might be that (at least in future) you will skip fields  of which you are not aware (e.g. now you forgot about IsHUPrice and M_HU_PI_Item_Product_ID).

Reply:
Great!
Thx! I wasn't aware of it!